### ISSUE TITLE
"Invalid operation at this state." Upon Connection

### ISSUE THREAD

**Issue by D1G1T4L3CH0** [DeathCradle/Terraria-s-Dedicated-Server-Mod#136]
serverconfig.txt:
`password=`

A Player joins the server and sees the world and character, then immediately gets disconnected with the following message.
`Invalid operation at this state.`

server.log:
```
7/12/2015 1:08:33 AM Command> Ready for commands.
7/12/2015 1:08:39 AM Pool> 127.0.0.1:49754 is connecting...
7/12/2015 1:08:39 AM Pool> Log message repeated 1 times
7/12/2015 1:08:39 AM Pool> ArgsPool<RecvArgs> capacity is now: 1.
7/12/2015 1:08:39 AM Pool> ArgsPool<SendArgs> capacity is now: 1.
7/12/2015 1:08:39 AM Pool> <Restrict> Letting user D1G1T4L3CH0 from 127.0.0.1 in as guest.
7/12/2015 1:08:39 AM Pool> Log message repeated 8 times
7/12/2015 1:08:39 AM Pool> 127.0.0.1:49754 was booted: Invalid operation at this state.
7/12/2015 1:08:39 AM Pool> 127.0.0.1:49754: connection closed (ConnectionAborted).
```

Nevermind the fact that you see restrict messages there, I tried also without restrict.
When restrict is loaded and a registered restrict user joins with the right password, they get in and can play.

**Comment by DeathCradle** [comment]
Should be fixed in f9b2790b0ebf30dd7ea814b06a4084d122be6ef2

**Comment by D1G1T4L3CH0** [comment]
resolved
